go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -892,8 +892,13 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 			return h.oc.addUpdateLocalNodeEvent(newNode, nodeSyncsParam)
 		} else {
 			_, syncZoneIC := h.oc.syncZoneICFailed.Load(newNode.Name)
-			// Check if the node moved from local zone to remote zone and if so syncZoneIC should be set to true
-			syncZoneIC = syncZoneIC || h.oc.isLocalZoneNode(oldNode)
+			// Check if the node moved from local zone to remote zone and if so syncZoneIC should be set to true.
+			// Also check if node subnet changed, so static routes are properly set
+			syncZoneIC = syncZoneIC || h.oc.isLocalZoneNode(oldNode) || nodeSubnetChanged(oldNode, newNode)
```

Comment:
Update: this is great, but not enough! It does not take into consideration cases where the node annotation set via [zone_cluster_controller](https://github.com/ovn-org/ovn-kubernetes/blob/86f9e1fd9439527eb12b3661c7e4feda311ab217/go-controller/pkg/clustermanager/zone_cluster_controller.go#L216)  see `node.Annotations[ovnTransitSwitchPortAddr]` changed.

See: https://github.com/ovn-org/ovn-kubernetes/issues/3770